Added 2 negative tests to CPU offline/online test case

juhlee-microsoft opened the pull request:
1. Removed the function reset_cpu_id and merge it to the main
2. Added 2 negative tests

**TEST RESULTS**
[LISAv2 Test Results Summary]
Test Run On           : 05/22/2020 19:50:09
ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
Initial Kernel Version: 4.15.0-1082-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:43

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                41.41 


[LISAv2 Test Results Summary]
Test Run On           : 05/22/2020 19:50:13
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Initial Kernel Version: 5.3.0-1020-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:26

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                23.43 


[LISAv2 Test Results Summary]
Test Run On           : 05/22/2020 20:34:29
ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
Initial Kernel Version: 4.15.0-1082-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:10

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS                66.86 


[LISAv2 Test Results Summary]
Test Run On           : 05/22/2020 20:34:35
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Initial Kernel Version: 5.3.0-1020-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:32

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 STRESS-CPU-OFFLINE-ONLINE

iamshital commented:
Execution is verified
---------------------

**ResourceCleanup = Delete**
```
02/27/2019 12:35:11 : [INFO ] CURRENT - PASS    - 3
02/27/2019 12:35:11 : [INFO ] CURRENT - SKIPPED - 0
02/27/2019 12:35:11 : [INFO ] CURRENT - FAIL    - 0
02/27/2019 12:35:11 : [INFO ] CURRENT - ABORTED - 0
02/27/2019 12:35:11 : [INFO ] ResourceCleanup is set to 'Delete'. Starting resource cleanup...
02/27/2019 12:35:11 : [WARN ] DeployVMPerEachTest=false will be ignored. (Conflicting parameter ResourceCleanup=Delete)
02/27/2019 12:35:11 : [INFO ] Try to delete resource group LISAv2-OneVM-SS-BV81-20190227043235...
02/27/2019 12:35:11 : [INFO ] Checking if LISAv2-OneVM-SS-BV81-20190227043235 exists...
02/27/2019 12:35:12 : [INFO ] Triggering : Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235...
02/27/2019 12:35:12 : [INFO ] Successfully delete resource group LISAv2-OneVM-SS-BV81-20190227043235..
02/27/2019 12:35:12 : [INFO ] BVT-VERIFY-BOOT-ERROR-WARNINGS-3 ended running with status: PASS.
02/27/2019 12:35:12 : [INFO ] Checking background cleanup jobs..
02/27/2019 12:35:12 : [INFO ] LISAv2-OneVM-SS-BV81-20190227042608 : Delete : Completed
02/27/2019 12:35:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227042941 is running.
02/27/2019 12:35:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:35:12 : [INFO ] 2 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:35:42 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227042941 is running.
02/27/2019 12:35:42 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:35:42 : [INFO ] 2 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:36:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227042941 is running.
02/27/2019 12:36:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:36:12 : [INFO ] 2 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:36:42 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227042941 is running.
02/27/2019 12:36:42 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:36:42 : [INFO ] 2 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:37:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227042941 is running.
02/27/2019 12:37:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:37:12 : [INFO ] 2 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:37:42 : [INFO ] LISAv2-OneVM-SS-BV81-20190227042941 : Delete : Completed
02/27/2019 12:37:42 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:37:42 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:38:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:38:12 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:38:42 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:38:42 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:39:12 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:39:12 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:39:42 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-BV81-20190227043235 is running.
02/27/2019 12:39:42 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:40:12 : [INFO ] LISAv2-OneVM-SS-BV81-20190227043235 : Delete : Completed
02/27/2019 12:40:12 : [INFO ] All background cleanup jobs finished.
02/27/2019 12:40:12 : [INFO ] Removed 0 context files.
02/27/2019 12:40:12 : [INFO ] Test BV81 finished
02/27/2019 12:40:12 : [INFO ]
[LISAv2 Test Results Summary]
Test Run On           : 02/27/2019 12:26:01
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Total Test Cases      : 1 (3 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:14

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-BOOT-ERROR-WARNINGS-1                                                  PASS                 0.81
    2 BVT-VERIFY-BOOT-ERROR-WARNINGS-2                                                  PASS                  0.7
    3 BVT-VERIFY-BOOT-ERROR-WARNINGS-3                                                  PASS                 0.67


Logs can be found at C:\GitHub\LISAv2\TestResults\2019-27-02-04-25-56-0055


02/27/2019 12:40:12 : [INFO ] Current Git Branch:  fix719
02/27/2019 12:40:12 : [INFO ] Current Git Commit ID: e91901c6d159336e1cf0828a061746bb39bffa29
02/27/2019 12:40:13 : [INFO ] Creating 'C:\GitHub\LISAv2\Azure-BV81-TestLogs.zip' from 'C:\GitHub\LISAv2\TestResults\2019-27-02-04-25-56-0055'
02/27/2019 12:40:13 : [INFO ] C:\GitHub\LISAv2\Azure-BV81-TestLogs.zip created successfully.
02/27/2019 12:40:13 : [INFO ] Analyzing results..
02/27/2019 12:40:13 : [INFO ] LISAv2 exit code: 0
```
**ResourceCleanup = Default**
```
02/27/2019 12:30:37 : [INFO ] CURRENT - PASS    - 3
02/27/2019 12:30:37 : [INFO ] CURRENT - SKIPPED - 0
02/27/2019 12:30:37 : [INFO ] CURRENT - FAIL    - 0
02/27/2019 12:30:37 : [INFO ] CURRENT - ABORTED - 0
02/27/2019 12:30:37 : [INFO ] BVT-VERIFY-BOOT-ERROR-WARNINGS-3 ended running with status: PASS.
02/27/2019 12:30:37 : [INFO ] Try to delete resource group LISAv2-OneVM-SS-TW75-20190227042541...
02/27/2019 12:30:37 : [INFO ] Checking if LISAv2-OneVM-SS-TW75-20190227042541 exists...
02/27/2019 12:30:37 : [INFO ] Triggering : Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541...
02/27/2019 12:30:37 : [INFO ] Successfully delete resource group LISAv2-OneVM-SS-TW75-20190227042541..
02/27/2019 12:30:37 : [INFO ] Checking background cleanup jobs..
02/27/2019 12:30:37 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:30:38 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:31:08 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:31:08 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:31:38 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:31:38 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:32:08 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:32:08 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:32:38 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:32:38 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:33:08 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:33:08 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:33:38 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:33:38 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:34:08 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-TW75-20190227042541 is running.
02/27/2019 12:34:08 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:34:38 : [INFO ] LISAv2-OneVM-SS-TW75-20190227042541 : Delete : Completed
02/27/2019 12:34:38 : [INFO ] All background cleanup jobs finished.
02/27/2019 12:34:38 : [INFO ] Removed 0 context files.
02/27/2019 12:34:38 : [INFO ] Test TW75 finished
02/27/2019 12:34:38 : [INFO ]
[LISAv2 Test Results Summary]
Test Run On           : 02/27/2019 12:25:35
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Total Test Cases      : 1 (3 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:9

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-BOOT-ERROR-WARNINGS-1                                                  PASS                 0.73
    2 BVT-VERIFY-BOOT-ERROR-WARNINGS-2                                                  PASS                 0.69
    3 BVT-VERIFY-BOOT-ERROR-WARNINGS-3                                                  PASS                 0.72


Logs can be found at C:\GitHub\LISAv2\TestResults\2019-27-02-04-25-30-1485


02/27/2019 12:34:38 : [INFO ] Current Git Branch:  fix719
02/27/2019 12:34:38 : [INFO ] Current Git Commit ID: e91901c6d159336e1cf0828a061746bb39bffa29
02/27/2019 12:34:38 : [INFO ] Creating 'C:\GitHub\LISAv2\Azure-TW75-TestLogs.zip' from 'C:\GitHub\LISAv2\TestResults\2019-27-02-04-25-30-1485'
02/27/2019 12:34:38 : [INFO ] C:\GitHub\LISAv2\Azure-TW75-TestLogs.zip created successfully.
02/27/2019 12:34:38 : [INFO ] Analyzing results..
02/27/2019 12:34:38 : [INFO ] LISAv2 exit code: 0
```
**ResourceCleanup = Keep**
```
02/27/2019 12:31:04 : [INFO ] CURRENT - PASS    - 3
02/27/2019 12:31:04 : [INFO ] CURRENT - SKIPPED - 0
02/27/2019 12:31:04 : [INFO ] CURRENT - FAIL    - 0
02/27/2019 12:31:04 : [INFO ] CURRENT - ABORTED - 0
02/27/2019 12:31:04 : [INFO ] BVT-VERIFY-BOOT-ERROR-WARNINGS-3 ended running with status: PASS.
02/27/2019 12:31:04 : [INFO ] Checking background cleanup jobs..
02/27/2019 12:31:04 : [INFO ] All background cleanup jobs finished.
02/27/2019 12:31:05 : [INFO ] Removed 0 context files.
02/27/2019 12:31:05 : [INFO ] Test WM52 finished
02/27/2019 12:31:05 : [INFO ]
[LISAv2 Test Results Summary]
Test Run On           : 02/27/2019 12:26:12
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Total Test Cases      : 1 (3 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:4

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-BOOT-ERROR-WARNINGS-1                                                  PASS                 0.78
    2 BVT-VERIFY-BOOT-ERROR-WARNINGS-2                                                  PASS                 0.75
    3 BVT-VERIFY-BOOT-ERROR-WARNINGS-3                                                  PASS                 0.64


Logs can be found at C:\GitHub\LISAv2\TestResults\2019-27-02-04-26-06-1415


02/27/2019 12:31:05 : [INFO ] Current Git Branch:  fix719
02/27/2019 12:31:05 : [INFO ] Current Git Commit ID: e91901c6d159336e1cf0828a061746bb39bffa29
02/27/2019 12:31:05 : [INFO ] Creating 'C:\GitHub\LISAv2\Azure-WM52-TestLogs.zip' from 'C:\GitHub\LISAv2\TestResults\2019-27-02-04-26-06-1415'
02/27/2019 12:31:05 : [INFO ] C:\GitHub\LISAv2\Azure-WM52-TestLogs.zip created successfully.
02/27/2019 12:31:05 : [INFO ] Analyzing results..
02/27/2019 12:31:05 : [INFO ] LISAv2 exit code: 0
```
**ResourceCleanup = Empty**
```
02/27/2019 12:31:32 : [INFO ] CURRENT - PASS    - 3
02/27/2019 12:31:32 : [INFO ] CURRENT - SKIPPED - 0
02/27/2019 12:31:32 : [INFO ] CURRENT - FAIL    - 0
02/27/2019 12:31:32 : [INFO ] CURRENT - ABORTED - 0
02/27/2019 12:31:32 : [INFO ] BVT-VERIFY-BOOT-ERROR-WARNINGS-3 ended running with status: PASS.
02/27/2019 12:31:32 : [INFO ] Try to delete resource group LISAv2-OneVM-SS-XK43-20190227042638...
02/27/2019 12:31:32 : [INFO ] Checking if LISAv2-OneVM-SS-XK43-20190227042638 exists...
02/27/2019 12:31:33 : [INFO ] Triggering : Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638...
02/27/2019 12:31:34 : [INFO ] Successfully delete resource group LISAv2-OneVM-SS-XK43-20190227042638..
02/27/2019 12:31:34 : [INFO ] Checking background cleanup jobs..
02/27/2019 12:31:34 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:31:34 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:32:04 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:32:04 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:32:34 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:32:34 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:33:04 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:33:04 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:33:34 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:33:34 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:34:04 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:34:04 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:34:34 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:34:34 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:35:04 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-XK43-20190227042638 is running.
02/27/2019 12:35:04 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/27/2019 12:35:34 : [INFO ] LISAv2-OneVM-SS-XK43-20190227042638 : Delete : Completed
02/27/2019 12:35:34 : [INFO ] All background cleanup jobs finished.
02/27/2019 12:35:34 : [INFO ] Removed 0 context files.
02/27/2019 12:35:34 : [INFO ] Test XK43 finished
02/27/2019 12:35:34 : [INFO ]
[LISAv2 Test Results Summary]
Test Run On           : 02/27/2019 12:26:32
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Total Test Cases      : 1 (3 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:9

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-BOOT-ERROR-WARNINGS-1                                                  PASS                  0.9
    2 BVT-VERIFY-BOOT-ERROR-WARNINGS-2                                                  PASS                 0.78
    3 BVT-VERIFY-BOOT-ERROR-WARNINGS-3                                                  PASS                 0.69


Logs can be found at C:\GitHub\LISAv2\TestResults\2019-27-02-04-26-26-5406


02/27/2019 12:35:34 : [INFO ] Current Git Branch:  fix719
02/27/2019 12:35:34 : [INFO ] Current Git Commit ID: e91901c6d159336e1cf0828a061746bb39bffa29
02/27/2019 12:35:34 : [INFO ] Creating 'C:\GitHub\LISAv2\Azure-XK43-TestLogs.zip' from 'C:\GitHub\LISAv2\TestResults\2019-27-02-04-26-26-5406'
02/27/2019 12:35:34 : [INFO ] C:\GitHub\LISAv2\Azure-XK43-TestLogs.zip created successfully.
02/27/2019 12:35:34 : [INFO ] Analyzing results..
02/27/2019 12:35:34 : [INFO ] LISAv2 exit code: 0
```

juhlee-microsoft commented:
@iamshital Please help me understand your test result above was not actual removal, correct?

juhlee-microsoft commented on TestControllers/TestController.psm1:
```diff
@@ -518,25 +518,36 @@ Class TestController
+							Write-LogWarn "DeployVMPerEachTest=false will be ignored. (Conflicting parameter ResourceCleanup=Delete)"
+						}
+						$this.TestProvider.DeleteTestVMS($vmData, $this.SetupTypeTable[$setupType], $this.UseExistingRG)
+					} elseif (!$this.TestCasePassStatus.contains($lastResult.TestResult)) {
```
The logic in if-statement does not make sense.  What if TestCasePassStatus contains TestResult AND ResourceCleanup is 'Delete', if goes to the first if-statement block. It never reach out to the second block. Please re-design the conditional logic.

iamshital replied:
@juhlee-microsoft  This code is unchanged. Please check the PR again.

I've added only one condition above this - to check If ResourceCleanup is equal to Delete or not. If yes, then we will continue to delete the and we don't care if the test result was PASS/FAIL/SKIPPED/ABORTED.

juhlee-microsoft replied:
@iamshital I am leaving my comment in your change of before-and-after comparison page

iamshital replied:
@juhlee-microsoft `if(!$this.TestCasePassStatus.contains($lastResult.TestResult))` check was already there. I didn't add it / remove it / change it.

![image](https://user-images.githubusercontent.com/7985635/53511210-deb77e80-3ae5-11e9-9dea-b9b9de6ce81d.png)

juhlee-microsoft replied:
You are right. but your new condition is placed before the original condition. if ResourceCleanup is matching to 'Delete', then it never reach to the second condition. I am curious if this is right design of the logic. Leave this with @lizzha

juhlee-microsoft replied:
Also this is a matter of Azure platform not others. Why did you put into common area?

iamshital commented:
> @iamshital Please help me understand your test result above was not actual removal, correct? It is nice to see the actual command of each result.

It is actual execution logs.

**Commands :** 

.\Run-LisaV2.ps1 -XMLSecretFile "XXXXXXXXXXXXXXXXXX.xml"  -TestPlatform Azure -TestLocation westus2 -RGIdentifier SS -ARMImageName 'Canonical UbuntuServer 16.04-LTS latest' -TestNames 'BVT-VERIFY-DEPLOYMENT-PROVISION' -ResourceCleanup Delete -TestIterations 3

.\Run-LisaV2.ps1 -XMLSecretFile "XXXXXXXXXXXXXXXXXX.xml"  -TestPlatform Azure -TestLocation westus2 -RGIdentifier SS -ARMImageName 'Canonical UbuntuServer 16.04-LTS latest' -TestNames 'BVT-VERIFY-DEPLOYMENT-PROVISION' -ResourceCleanup Default -TestIterations 3

.\Run-LisaV2.ps1 -XMLSecretFile "XXXXXXXXXXXXXXXXXX.xml"  -TestPlatform Azure -TestLocation westus2 -RGIdentifier SS -ARMImageName 'Canonical UbuntuServer 16.04-LTS latest' -TestNames 'BVT-VERIFY-DEPLOYMENT-PROVISION' -ResourceCleanup Keep -TestIterations 3

.\Run-LisaV2.ps1 -XMLSecretFile "XXXXXXXXXXXXXXXXXX.xml"  -TestPlatform Azure -TestLocation westus2 -RGIdentifier SS -ARMImageName 'Canonical UbuntuServer 16.04-LTS latest' -TestNames 'BVT-VERIFY-DEPLOYMENT-PROVISION' -TestIterations 3

.\Run-LisaV2.ps1 -XMLSecretFile "XXXXXXXXXXXXXXXXXX.xml"  -TestPlatform Azure -TestLocation westus2 -RGIdentifier SS -ARMImageName 'Canonical UbuntuServer 16.04-LTS latest' -TestNames 'BVT-VERIFY-DEPLOYMENT-PROVISION' -TestIterations 3 -DeployVMPerEachTest

lizzha commented on TestControllers/TestController.psm1:
```diff
@@ -518,25 +518,36 @@ Class TestController
-							$this.TestProvider.DeleteTestVMS($vmData, $this.SetupTypeTable[$setupType], $this.UseExistingRG)
-						} elseif (!$this.TestProvider.ReuseVmOnFailure) {
+
+					if ($this.ResourceCleanup -imatch "Delete") {
```
It violates the design, that if the test pass, then the VM can be reused for next test case. If the test pass, then there's no need to delete the VM (unless DeployVMPerEachTest is set, or it's the last case of same deployment), even ResourceCleanup is set to Delete. 
I'm not sure whether it's the root cause of the issue, the previous logic checks when Test fails/aborts, and if ResourceCleanup is set to Delete, it will delete the VM.
![image](https://user-images.githubusercontent.com/7566196/53535399-a3489e80-3b3d-11e9-95df-5047ebce29c5.png)

lizzha commented:
Please check email, the actual issue is that resource group is not deleted when azure deployment fails

iamshital commented:
@lizzha Understood. 
Based on the new comments, #826 fixes "TestResult = PASS && ResourceCleanup = Delete" issue.
Will update the current PR to accommodate this change.

iamshital commented:
@lizzha  Updated the PR. **Logs**

```
02/28/2019 11:59:30 : [INFO ] Creating Deployment using C:\Users\v-prmhet\AppData\Local\Temp\LISAv2-OneVM-SS-QV96-20190228035916.json ...
VERBOSE: Performing the operation "Creating Deployment" on target "LISAv2-OneVM-SS-QV96-20190228035916".
VERBOSE: 3:59:33 AM - Template is valid.
VERBOSE: 3:59:36 AM - Create template deployment 'eosg636869231703137993'
VERBOSE: 3:59:36 AM - Checking deployment status in 5 seconds
VERBOSE: 3:59:41 AM - Resource Microsoft.Storage/storageAccounts 'eosgwestus2' provisioning status is succeeded
VERBOSE: 3:59:41 AM - Resource Microsoft.Compute/availabilitySets 'AvailibilitySet' provisioning status is succeeded
VERBOSE: 3:59:41 AM - Checking deployment status in 5 seconds
VERBOSE: 3:59:46 AM - Resource Microsoft.Network/loadBalancers 'LoadBalancer' provisioning status is succeeded
VERBOSE: 3:59:46 AM - Resource Microsoft.Network/publicIPAddresses 'PublicIPv4-87355' provisioning status is succeeded
VERBOSE: 3:59:46 AM - Resource Microsoft.Network/virtualNetworks 'VirtualNetwork' provisioning status is succeeded
VERBOSE: 3:59:46 AM - Checking deployment status in 5 seconds
VERBOSE: 3:59:51 AM - Resource Microsoft.Network/networkInterfaces 'PrimaryNIC-LISAv2-OneVM-SS-QV96-20190228035916-role-0' provisioning status is succeeded
VERBOSE: 3:59:51 AM - Checking deployment status in 5 seconds
VERBOSE: 3:59:56 AM - Checking deployment status in 5 seconds
02/28/2019 12:00:02 : [ERROR] Failed to create Resource Group Deployment - LISAv2-OneVM-SS-QV96-20190228035916.
02/28/2019 12:00:02 : [INFO ] -ResourceCleanup = Delete is Set. Deleting LISAv2-OneVM-SS-QV96-20190228035916.
02/28/2019 12:00:02 : [INFO ] Try to delete resource group LISAv2-OneVM-SS-QV96-20190228035916...
02/28/2019 12:00:02 : [INFO ] Checking if LISAv2-OneVM-SS-QV96-20190228035916 exists...
02/28/2019 12:00:02 : [INFO ] Triggering : Delete-ResourceGroup-LISAv2-OneVM-SS-QV96-20190228035916...
02/28/2019 12:00:03 : [INFO ] Cleanup successful for LISAv2-OneVM-SS-QV96-20190228035916.
02/28/2019 12:00:03 : [ERROR] Unable to Deploy one or more VM's
02/28/2019 12:00:03 : [ERROR] One or More Deployments are Failed..!
02/28/2019 12:00:03 : [INFO ] CURRENT - PASS    - 0
02/28/2019 12:00:03 : [INFO ] CURRENT - SKIPPED - 0
02/28/2019 12:00:03 : [INFO ] CURRENT - FAIL    - 0
02/28/2019 12:00:03 : [INFO ] CURRENT - ABORTED - 1
02/28/2019 12:00:03 : [INFO ] Checking background cleanup jobs..
02/28/2019 12:00:03 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-QV96-20190228035916 is running.
02/28/2019 12:00:03 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/28/2019 12:00:33 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-QV96-20190228035916 is running.
02/28/2019 12:00:33 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/28/2019 12:01:03 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-QV96-20190228035916 is running.
02/28/2019 12:01:03 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/28/2019 12:01:33 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-QV96-20190228035916 is running.
02/28/2019 12:01:33 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/28/2019 12:02:03 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-QV96-20190228035916 is running.
02/28/2019 12:02:03 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/28/2019 12:02:33 : [INFO ] Delete-ResourceGroup-LISAv2-OneVM-SS-QV96-20190228035916 is running.
02/28/2019 12:02:33 : [INFO ] 1 background cleanup jobs still running. Waiting 30 seconds...
02/28/2019 12:03:03 : [INFO ] LISAv2-OneVM-SS-QV96-20190228035916 : Delete : Completed
02/28/2019 12:03:03 : [INFO ] All background cleanup jobs finished.
02/28/2019 12:03:03 : [INFO ] Removed 0 context files.
02/28/2019 12:03:03 : [INFO ] Test QV96 finished
02/28/2019 12:03:03 : [INFO ]
[LISAv2 Test Results Summary]
Test Run On           : 02/28/2019 11:59:10
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest2
Total Test Cases      : 1 (0 Passed, 0 Failed, 1 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:3

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-BOOT-ERROR-WARNINGS                                                 Aborted                    0


Logs can be found at C:\GitHub\LISAv2\TestResults\2019-28-02-03-59-04-0397


02/28/2019 12:03:03 : [INFO ] Current Git Branch:  fix719
02/28/2019 12:03:03 : [INFO ] Current Git Commit ID: 702fc175c0d5f314c20c1fde0ced92864ca276c5
02/28/2019 12:03:03 : [INFO ] Creating 'C:\GitHub\LISAv2\Azure-QV96-TestLogs.zip' from 'C:\GitHub\LISAv2\TestResults\2019-28-02-03-59-04-0397'
02/28/2019 12:03:03 : [INFO ] C:\GitHub\LISAv2\Azure-QV96-TestLogs.zip created successfully.
02/28/2019 12:03:03 : [INFO ] Analyzing results..
02/28/2019 12:03:03 : [INFO ] LISAv2 exit code: 1
```

juhlee-microsoft commented on Libraries/Azure.psm1:
```diff
@@ -542,7 +542,7 @@ Function Create-ResourceGroupDeployment([string]$RGName, $location, $TemplateFil
                     }
                     else {
                         Write-LogInfo "Removing Failed resource group, as we found 0 VM(s) deployed."
-                        $isCleaned = Delete-ResourceGroup -RGName $RGName -UseExistingRG
```
This is interesting idea, @LiliDeng @lizzha  can you review this removal?

lizzha replied:
why removing the parameter "-UseExistingRG"?

iamshital replied:
@lizzha 
This command was failing with exception : no value provided to the parameter 
UseExistingRG. 

Function Delete-ResourceGroup accepts "boolean" parameter, where as it appears that, we tried to use as "switch" parameter. I believe this is human error?

Please check the fixed code.

lizzha replied:
yes, it was error. thanks for the fix

LiliDeng commented:
Click re-request by mistake, please ignore @juhlee-microsoft

juhlee-microsoft committed:
Added 2 negative tests

1. Removed the function reset_cpu_id and merge it to the main
2. Added 2 negative tests

juhlee-microsoft committed:
Added Failed Case count for final verification

juhlee-microsoft committed:
Removed one of reset module becasue it depends on the other vmbus dependency.

juhlee-microsoft committed:
Fixed per comment
